ratatool-scalacheck/README.md

```diff
@@ -39,7 +37,50 @@ Implicit Arbitrary instances are also available for Avro and Protobuf records. E
 
 ```scala
 val avroArb: Arbitrary[MyRecord] = implicitly[Arbitrary[MyRecord]]
+```
+
+Multiple `Gen` instances can be composed together with a for-comprehension. This will cause the data to be generated with the same seed.
+```scala
+val genData: Gen[(MyRecord, Int)] = for {
+ record <- avroGen
+ i <- Gen.choose(0, 10)
+} yield (record, i)
+
+// .sample gets a random seed
+val data: Option[(MyRecord, Int)] = genData.sample
+```
+
+Due to scalacheck internals generation can fail so the return type of `.sample` is `Option[A]`.
+In general, calling `.get` on an `Option` is an anti-pattern because this can produce `NoSuchElementException`.
```

Comment:
This prescriptive tone was motivated by the substantial misuse of `Gen` we've seen internally, but I can definitely tone it down.

Reply:
I'm not sure I wholly agree with your definition of `misuse`. In some cases yes, but other times people do things like `.sample.get` because it's easier and they don't need such strong guarantees. There's nothing explicitly wrong with that, we just need to make sure they are capable of making an informed decision.